crates/polars-plan/src/client/check.rs

```diff
@@ -39,23 +24,7 @@ pub(super) fn assert_cloud_eligible(dsl: &DslPlan) -> PolarsResult<()> {
                     return ineligible_error("contains sink to non-cloud location");
                 }
             },
-            plan => {
-                plan.get_expr(&mut expr_stack);
-
-                for expr in expr_stack.drain(..) {
-                    for expr_node in expr.into_iter() {
-                        match expr_node {
-                            Expr::AnonymousFunction { .. } => {
```

Comment:
This isn't sufficient. `AnonymousFunction` isn't completely serde. I think we should extend the `dyn GetOutput` with a `try_serialize`, like we did for the for input function `dyn SeriesUdf` .


After that we can implement `serde` similarly. The `try_serialize` ensures we still fail client side if we cannot serialize the udf.

Reply:
> `AnonymousFunction` isn't completely serde.

Do you have an example of a UDF that would pass the current serialization but would not deserialize correctly on the server? Then I can test for it. I don't really know what the conditions would be for that to occur.

Reply:
The function doesn't matter per se. The output dtype matters. So any `map_elements` where you set `return_dtype` will deserialize to the default `GetOutput` impl.